src/main/java/opentree/plugins/tree_of_life.java

```diff
@@ -300,15 +311,25 @@ public Representation induced_subtree (@Source GraphDatabaseService graphDb,
 
         */
 
-        Map<String, Object> result = v3.doInducedSubtree(graphDb, longIdsToStringIds(nodeIDs), ottIDs, "name_and_id", Boolean.FALSE);
-        result.put("newick", result.get("newick"));
-        result.put("node_ids_not_in_tree", empty);
-        result.put("node_ids_not_in_graph", empty);
-        result.put("ott_ids_not_in_tree", empty);
-        result.put("ott_ids_not_in_graph", empty);
-        result.put("tree_id", treeId);
+        Map<String, Object> v2_result = new HashMap<>(); // the result we're returning
+        Map<String, Object> v3_result; // the result returned by the v3 method
+        try {
+            v3_result = v3.doInducedSubtree(graphDb, longIdsToStringIds(nodeIDs), ottIDs, "name_and_id", Boolean.FALSE);
+            v2_result.put("node_ids_not_in_tree", empty);
+            v2_result.put("ott_ids_not_in_tree", empty);
+        } catch (BadIdsException e) {
+            v3_result = e.json;
+            v2_result.put("node_ids_not_in_tree", stringIdsToLongIds(e.nodeIds));
+            v2_result.put("ott_ids_not_in_tree", e.ottIds);
+        }
 
-        return OTRepresentationConverter.convert(result);
+        v2_result.put("newick", v3_result.get("newick"));
```

Comment:
I see that the v3_result is sort of incorporated into v2_result, but do we ever simply return v3_result?